svg.js.d.ts

```diff
@@ -144,46 +614,421 @@ declare module "@svgdotjs/svg.js" {
         h: number;
         x2: number;
         y2: number;
-        merge(box: Box): Box;
-        transform(m: Matrix): Box
+
+        constructor();
         constructor(source: string);
-        constructor(source: []);
-        constructor(source: object);
+        constructor(source: number[]);
+        constructor(source: DOMRect);
         constructor(x: number, y: number, width: number, height: number);
+
+        merge(box: Box): Box;
+        transform(m: Matrix): Box
         addOffset(): this;
         toString(): string;
         toArray(): number[];
         isNulled(): boolean;
+        to(v: MorphValueLike): Morphable;
     }
 
-    // clip.js
-    class ClipPath extends Container {
+    // Morphable.js
+    type MorphValueLike = string | number | ObjectBag | NonMorphable | MatrixExtract | Array<any> | any[]
+
+    class Morphable {
         constructor();
-        targets: Element[];
-        remove(): this;
+        constructor(st: Stepper);
+
+        from(): MorphValueLike
+        from(v: MorphValueLike): this
+        to(): MorphValueLike
+        to(v: MorphValueLike): this
+        type(): any
+        type(t: any): this
+        stepper(): Stepper
+        stepper(st: Stepper): this
+        done(): boolean
+        at(pos: number): any
     }
 
-    // color.js
-    interface ColorLike {
-        r: number;
-        g: number;
-        b: number;
+    class ObjectBag {
+        constructor();
+        constructor(a: Object);
+        valueOf(): Object
+        toArray(): Object[]
+
+        to(a: Object): Morphable
+        fromArray(a: Object): this
     }
 
-    type ColorAlias = string | ColorLike;
+    class NonMorphable {
+        constructor(a: Object)
+        valueOf(): Object
+        toArray(): Object[]
+
+        to(a: Object): Morphable
+        fromArray(a: Object): this
+    }
+
+    // TODO: check & verify ??????
+    class TransformBag {
+        constructor()
+        constructor(a: number[])
+        constructor(a: TransformData)
+
+        defaults: TransformData
+
+        toArray(): number[]
+
+        to(t: TransformData): Morphable
+        fromArray(t: TransformData): this

```

Comment:
fromArray takes an array as input. Beside that, this looks fine

Reply:
would that be an array of numbers?

Reply:
yes!